Add document for robot account (#7111)
goharbor/harbor: docs/user_guide.md

> In Harbor portal, enter your project, select the repository, click on the link of tag name you'd like to see its build history, the detail page will be opened. Then switch to `Build History` tab, you can see the build history information.

Reviewer: enter your project -> enter the project

Author: it's not in this PR....

![build_ history](img/build_history.png)

## Robot Account

> Robot accounts are accounts created by project admin that are intended for automated operations, 

Reviewer 2: ```suggestion
Robot Accounts are accounts created by project admins that are intended for automated operations. They have the following limitations:
```

> 1, Robot account cannot login Harbor portal.

Reviewer 2: ```suggestion
1. Robot Accounts cannot login Harbor portal
```

> 1, Robot Accounts cannot login Harbor portal

Reviewer: add a period to the end.

> 2, Robot account can do docker pull/push only with token.

Reviewer 2: ```suggestion
2. Robot Accounts only perform `docker push`/`docker pull` operations with a token
```

> ### Add a robot account

Reviewer 2: Try to keep the casing consistent:

```suggestion
### Add a Robot Account
```

Reviewer 2: There are a few other places further down you may want to check as well

> If you are a project admin, you can create a robot account by clicking "New Robot Account" in the **'Robot Accounts'** tab of a project, and enter a name, a description and permission.

Reviewer 2: ```suggestion
If you are a project admin, you can create a robot account by clicking "New Robot Account" in the `Robot Accounts` tab of a project, and enter a name, a description and permission.
```

![add_robot_account](img/robotaccount/add_robot_account.png)

![add_robot_account](img/robotaccount/add_robot_account_2.png)

> **NOTES:** The name will become robot$accountname that used to distinguish a robot account from harbor user.

Reviewer 2: ```suggestion
> **NOTE:** The name will become `robot$<accountname>` and will be used to distinguish a robot account from a normal harbor user.
```

![copy_robot_account_token](img/robotaccount/copy_robot_account_token.png)

> As Harbor doesn't store your account token, please make sure to copy it in the pop up dialog after creating, otherwise, there is no way to get it from Harbor.

Reviewer 2: Is this actually the case? Or do we just not show it to the user after it's generated? I'd think it'd *have* to be stored to verify the robot during `push`/`pull` operations...

Author: Per security concern, Harbor doesn't store the token of robot accounts. The only way to get it is from the pop-up dialog.

Reviewer 2: Oh I see, it's a JWT token. Makes sense to me now!

### Configure duration of robot account

> If you are a system admin, you can configure the robot account token duration in days. 

Reviewer: token duration -> token time to live or token expiration

Author: `token duration` is an comment term when to talk about expiration. let's keep it.

![set_robot_account_token_duration](img/robotaccount/set_robot_account_token_duration.png)

### Authenticate with a robot account
To authenticate with a Robot Account, use `docker login` as below,

```
docker login harbor.io
Username: robot$accountname
Password: Thepasswordgeneratedbyprojectadmin
```

### Disable a robot account

> If you are a project admin, you can disable a Robot Account by clicking "Disable Account" in the `Robot Accounts` tab of a project.

Reviewer: For project admin, you can ....

Author: to keep the consistency of doc, can you just use `If you are a project admin`

![disable_robot_account](img/robotaccount/disable_delete_robot_account.png)

### Delete a robot account

> If you are a project admin, you can delete a Robot Account by clicking "Delete" in the `Robot Accounts` tab of a project.

Reviewer: For project admins,

Author: same
